### PR TITLE
No selfloading jnlp

### PR DESCRIPTION
**Pull request by sclassen**

### CODE REVIEW

File: core/src/main/java/net/adoptopenjdk/icedteaweb/client/parts/dialogs/security/apptrustwarningpanel/MatchingALACAttributePanel.java [hunk 1]
```diff
@@ -118,7 +119,7 @@ public static void main(String[] args) throws MalformedURLException, IOException
         Set<URL> s = new HashSet<>();
```
Comment (hendrikebbers) [inline core/src/main/java/net/adoptopenjdk/icedteaweb/client/parts/dialogs/security/apptrustwarningpanel/MatchingALACAttributePanel.java]: Can we just remove this unneeded main method?

File: core/src/main/java/net/adoptopenjdk/icedteaweb/client/parts/dialogs/security/apptrustwarningpanel/UnsignedAppletTrustWarningPanel.java [hunk 2]
```diff
@@ -107,7 +108,7 @@ protected String getQuestionPanelText() {
     }
     
     public static void main(String[] args) throws Exception {
-        UnsignedAppletTrustWarningPanel w = new UnsignedAppletTrustWarningPanel(null, new JNLPFile(new URL("http://www.geogebra.org/webstart/geogebra.jnlp")));
```
Comment (hendrikebbers) [inline core/src/main/java/net/adoptopenjdk/icedteaweb/client/parts/dialogs/security/apptrustwarningpanel/UnsignedAppletTrustWarningPanel.java]: Can we just remove this unneeded main method?

File: core/src/main/java/net/sourceforge/jnlp/JNLPFile.java [hunk 3]
```diff
@@ -223,72 +217,41 @@
...
-            // null values will still work, and app can set defaults later
+            // FIXME: how should we proceed if the default values are not available??
         }
     }
 
     /**
      * Empty stub, allowing child classes to override the constructor
      */
+    // only used for tests
     protected JNLPFile() {
```
Comment (hendrikebbers) [inline core/src/main/java/net/sourceforge/jnlp/JNLPFile.java]: why do we need this constructor?

File: core/src/main/java/net/sourceforge/jnlp/JNLPFileFactory.java [hunk 4]
```diff
@@ -22,16 +22,104 @@
...
+     *
+     * @param location the location of the JNLP file
+     * @param version  the version of the JNLP file
+     * @param settings the {@link ParserSettings} to use when parsing the {@code location}
+     * @param policy   the update policy
+     * @throws IOException    if an IO exception occurred
+     * @throws ParseException if the JNLP file was invalid
+     */
+    public JNLPFile create(final URL location, final VersionString version, final ParserSettings settings, UpdatePolicy policy) throws IOException, ParseException {
+        final String uniqueKey = Calendar.getInstance().getTimeInMillis() + "-" + ((int) (Math.random() * Integer.MAX_VALUE)) + "-" + location;
```
Comment (hendrikebbers) [inline core/src/main/java/net/sourceforge/jnlp/JNLPFileFactory.java]: Why not a UUID?

File: core/src/main/java/net/sourceforge/jnlp/JNLPFile.java (hunk 3 above)
Comment (sclassen) [reply core/src/main/java/net/sourceforge/jnlp/JNLPFile.java]: there are dummy sub-classes in tests which call this constructors

File: core/src/main/java/net/sourceforge/jnlp/JNLPFileFactory.java (hunk 4 above)
Comment (sclassen) [reply core/src/main/java/net/sourceforge/jnlp/JNLPFileFactory.java]: I copied the existing code

**Pull request merged**